cirq/devices/two_qubit_noise_model.py

```diff
@@ -0,0 +1,840 @@
+                                                protocols.unitary(ZYGate())),
+                (self._p_zz,
+                 protocols.unitary(ZZGate())), (self._p_ix,
+                                                protocols.unitary(IXGate())),
+                (self._p_iy,
+                 protocols.unitary(IYGate())), (self._p_iz,
+                                                protocols.unitary(IZGate())))
+
+    def _has_mixture_(self) -> bool:
+        return True
+
+    def _value_equality_values_(self):
+        return self._p_xi, self._p_xx, self._p_xy, self._p_xz, \
+               self._p_yi, self._p_yx, self._p_yy, self._p_yz, \
+               self._p_zi, self._p_zx, self._p_zy, self._p_zz, \
+               self._p_iz, self._p_ix, self._p_iy
+
+    def __repr__(self) -> str:
+        return 'two_qubit_asymmetric_depolarize' \
+               '(p_xi={!r},p_xx={!r},p_xy={!r},p_xz={!r}, ' \
+               'p_yi={!r},p_yx={!r},p_yy={!r},p_yz={!r}, ' \
+               'p_zi={!r},p_zx={!r},p_zy={!r},p_zz={!r}, ' \
+               'p_ix={!r},p_iy={!r},p_iz={!r})'.format(
+            self._p_xi, self._p_xx, self._p_xy, self._p_xz,
+            self._p_yi, self._p_yx, self._p_yy, self._p_yz,
+            self._p_zi, self._p_zx, self._p_zy, self._p_zz,
+            self._p_iz, self._p_ix, self._p_iy)
+
+    def __str__(self) -> str:
+        return 'two_qubit_asymmetric_depolarize' \
+               '(p_xi={!r},p_xx={!r},p_xy={!r},p_xz={!r}, ' \
+               'p_yi={!r},p_yx={!r},p_yy={!r},p_yz={!r}, ' \
+               'p_zi={!r},p_zx={!r},p_zy={!r},p_zz={!r}, ' \
+               'p_ix={!r},p_iy={!r},p_iz={!r})'.format(
+            self._p_xi, self._p_xx, self._p_xy, self._p_xz,
+            self._p_yi, self._p_yx, self._p_yy, self._p_yz,
+            self._p_zi, self._p_zx, self._p_zy, self._p_zz,
+            self._p_iz, self._p_ix, self._p_iy)
+
+    def _circuit_diagram_info_(self,
+                               args: protocols.CircuitDiagramInfoArgs) -> str:
+        if args.precision is not None:
+            f = '{:.' + str(args.precision) + 'g}'
+            return 'A({},{},{},{},{},{},{},{},{},{},{},{},{},{},{},{})' \
+                .format(f, f, f, f, f, f, f, f, f, f, f, f, f, f, f, f, ).format(
+                        self._p_xi, self._p_xx, self._p_xy, self._p_xz,
+                        self._p_yi, self._p_yx, self._p_yy, self._p_yz,
+                        self._p_zi, self._p_zx, self._p_zy, self._p_zz,
+                        self._p_iz, self._p_ix, self._p_iy)
+        return 'A({!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r},{!r})'.format(
+            self._p_xi, self._p_xx, self._p_xy, self._p_xz, self._p_yi,
+            self._p_yx, self._p_yy, self._p_yz, self._p_zi, self._p_zx,
+            self._p_zy, self._p_zz, self._p_iz, self._p_ix, self._p_iy)
+
+
+def two_qubit_asymmetric_depolarize(p_xi: float, p_yi: float, p_zi: float,

```

Comment:
Add documentation.

Reply:
In the one qubit depolarising channel this isn't documented. I've added a short descripton. Should I also describe all parameters with `:param:`?

Reply:
> Should I also describe all parameters with :param:?

Yes please.